### ISSUE TITLE
LDAP Auth Backend Breaks Upon Upgrading from 2.0 to 2.1

### ISSUE THREAD

**Issue by jhull123** [StackStorm/st2#3213]
After following https://docs.stackstorm.com/install/upgrades.html to upgrade from St2 2.0 to 2.1 I found that the st2auth service would not start when the LDAP auth backend was configured.

```
2017-02-10 15:10:09,435 140229847622672 ERROR __init__ [-] Invalid authentication backend specified: ldap
Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2auth/backends/__init__.py", line 51, in get_backend_instance
    invoke_on_load=False)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/stevedore/driver.py", line 45, in __init__
    verify_requirements=verify_requirements,
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/stevedore/named.py", line 56, in __init__
    self._init_plugins(extensions)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/stevedore/driver.py", line 97, in _init_plugins
    (self.namespace, name))
RuntimeError: No 'st2auth.backends.backend' driver found, looking for 'ldap'
```

Doing the following fixed it.

```
st2ctl stop
yum remove st2-auth-ldap -y
yum install st2-auth-ldap bwc-enterprise -y
st2ctl start
```

**Comment by LindsayHill** [comment]
Thanks for the report @jhull123. It's a definite bug, and I'm pretty sure I've tracked down the packaging problem. 

The .spec file contains this:
```
%postun
  echo y | %{pip} uninstall st2-enterprise-auth-backend-ldap 1>/dev/null || :
```
The problem is that the `%postun` step gets run at the end of an upgrade, not just when doing an uninstall. 

It needs an additional hook in there to check if it's an upgrade or uninstall, and only run that step if it's the last package getting removed. I'll submit a PR for this.

Your workaround is OK. You could also re-run the postinst script with this:
```
/opt/stackstorm/st2/bin/pip install --find-links /opt/stackstorm/share/wheels --no-index --quiet st2-enterprise-auth-backend-ldap
```

**Comment by LindsayHill** [comment]
PRs have been submitted. Not sure if this will make v2.2 or not. May still need to execute manual workaround for v2.1.1 -> v2.2 upgrade. But it will definitely make whatever version comes after v2.2.

**Comment by LindsayHill** [comment]
This has been resolved with 2.2. Updated packages have been published.
